reassembler_test.go

```diff
@@ -150,10 +150,61 @@ func testReassembler(t testing.TB, file string, expected *results) {
 	}
 }
 
-func TestSequenceNumSliceSort(t *testing.T) {
-	expected := sequenceNumSlice{maxSeq - 5, maxSeq - 4, maxSeq - 3, maxSeq - 2, maxSeq, 0, 1, 2, 3, 4}
-	seqs := sequenceNumSlice{maxSeq - 5, maxSeq - 4, 0, 1, 2, maxSeq - 3, maxSeq - 2, maxSeq, 3, 4}
-	seqs.Sort()
+func Benchmark_eventList_put(b *testing.B) {
+	const maxSize = 10
+	h := &intHeap{}
+	heap.Init(h)
+	eventList := &eventList{
+		seqs:    h,
+		events:  make(map[int]*event, maxSize+1),
+		maxSize: maxSize,
+	}
+	rand.Seed(time.Now().Unix())
+
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		var msgType auparse.AuditMessageType
+		if i%2 == 0 {
+			msgType = auparse.AUDIT_EOE
+		}
+		eventList.Put(&auparse.AuditMessage{
+			Sequence:   rand.Uint32(),
+			RecordType: msgType,
+		})
+	}
+}
+
+func generateEvents() *eventList {
+	const maxSize = 10
+	h := &intHeap{}
+	heap.Init(h)
+	eventList := &eventList{
+		seqs:    h,
+		events:  make(map[int]*event, maxSize+1),
+		maxSize: maxSize,
+	}
+
+	for i := 0; i < 1000; i++ {
+		var msgType auparse.AuditMessageType
+		if i%2 == 0 {
+			msgType = auparse.AUDIT_EOE
+		}
+		eventList.Put(&auparse.AuditMessage{
+			Sequence:   rand.Uint32(),
+			RecordType: msgType,
+		})
+	}
+
+	return eventList
+}
+
+func Benchmark_eventList_cleanup(b *testing.B) {
+	eventList := generateEvents()
 
-	assert.Equal(t, expected, seqs)
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		eventList.CleanUp()
```

Comment:
it is useless to call CleanUp() in a loop like that. First call will eliminate all elements from eventList, and following will not do anything at all. I advise to move eventList initialization inside the loop, surrounded by calls b.StopTimer, b.StartTimer